### PR TITLE
[SYCL] Fix corner case: USM, 2nd CG depends_on 1st

### PR DESCRIPTION
**Pull request by dm-vodopyanov**
This patch fixes corner case where during USM, there are two CGs
from two different queues and contexts, and second CG `depends_on` first CG.
There are no memory dependencies between these two CGs. Without this patch,
kernel from second CG didn't run. This is happened because command containing
non-empty MPreparedHostDepsEvents was cleaned up, but shouldn't.

Test in intel/llvm-test-suite: https://github.com/intel/llvm-test-suite/pull/604

### CODE REVIEW

**Comment by steffenlarsen** [comment]
Looks good. Would you mind changing the comment on the success-case use of `CleanUp`? It does not seem in line with the changes.

**Review by sergey-semenov — CHANGES_REQUESTED** [review]
Discussed this change separately. It's not clear right now why the deletion of the first ExecCG command leads to any issues here, since its users should only refer to the event provided with depends_on rather than the command itself (since Event->MCommand is set to nullptr during deletion). I'm also worried that this change leaves us with memory leaks for commands with no  implicit dependencies if the user doesn't call queue/event::wait for them (which is what CleanUp() was initially supposed to address).

File: sycl/source/detail/scheduler/scheduler.cpp [hunk 1]
```diff
@@ -117,7 +117,12 @@ EventImplPtr Scheduler::addCG(std::unique_ptr<detail::CG> CommandGroup,
     bool Enqueued;
 
     auto CleanUp = [&]() {
-      if (NewCmd && (NewCmd->MDeps.size() == 0 && NewCmd->MUsers.size() == 0)) {
+      if (NewCmd && (NewCmd->MDeps.size() == 0 && NewCmd->MUsers.size() == 0 &&
+                     NewCmd->getPreparedHostDepsEvents().size() == 0)) {
+        // Calling assert to prevent cleaning up of the command which enqueue
+        // status is not successfull
+        assert(NewCmd->isSuccessfullyEnqueued());
```
Comment (steffenlarsen) [inline sycl/source/detail/scheduler/scheduler.cpp:124]: If this is a requirement for the cleanup to do anything, does it make sense to call it from the fail-cases? Most often they will report `EnqueueResultT::SyclEnqueueFailed` and thus not report the command as having been enqueued succesfully.

**Pull request closed without merging**